tools/fm-editor/tree/FeatureTreeItem.cpp

```diff
@@ -0,0 +1,101 @@
+//
+// Created by simon on 02.02.23.
+//
+#include "FeatureTreeItem.h"
+#include <QMenu>
+#include <unordered_set>
+#include "../Utils.h"
+QVariant numericValue(vara::feature::Feature* Item) {
+  if(Item->getKind()==vara::feature::Feature::FeatureKind::FK_NUMERIC){
+    auto *NumItem = dynamic_cast<vara::feature::NumericFeature*>(Item);
+    string Result ="[";
+    if(std::holds_alternative<vara::feature::NumericFeature::ValueRangeType>(NumItem->getValues())){
+      auto range = std::get<vara::feature::NumericFeature::ValueRangeType>(NumItem->getValues());
+      Result+= std::to_string(range.first) +", "+std::to_string(range.second) + "]";
+    } else {
+      auto Range = std::get<vara::feature::NumericFeature::ValueListType>(NumItem->getValues());
+      for(auto It = Range.begin();It!=Range.end();It++){
+        if(It!=Range.begin()){
+          Result+=",";
+        }
+        Result+=std::to_string(*It.base());
+      }
+      Result+="]";
+    }
+    return  QString::fromStdString(Result);
+  }
+  return {};
+}
+
+
+QVariant locationString(vara::feature::Feature* Item){
+  auto Locs = Item->getLocations();
+  std::stringstream StrS;
+  if (Item->hasLocations()) {
+    std::for_each(Locs.begin(), Locs.end(),
+                  [&StrS](const vara::feature::FeatureSourceRange &Fsr) {
+                    StrS << llvm::formatv("{0}", Fsr.toString()).str();
+                  });
+  }
+  return QString::fromStdString(StrS.str());
+}
+FeatureTreeItem* FeatureTreeItem::createFeatureTreeItem(
+    vara::feature::FeatureTreeNode *Item)  {
+  if(Item->getKind() == vara::feature::FeatureTreeNode::NodeKind::NK_RELATIONSHIP){
+    return new FeatureTreeItemRelation(dynamic_cast<vara::feature::Relationship*>(Item));
```

Comment:
dynamic casts with Relationship/Feature types is not a good idea, as they are implemented with the LLVM dyn_cast/classof pattern. You should use `dyn_cast<>` to cast them.